frontend-react/src/components/Admin/AdminReceiverDashboard.tsx

```diff
@@ -457,56 +500,165 @@ function MainRender(props: {
-                            <Grid row className={"week-column"}>
-                                {perDayElements}
-                            </Grid>
-                        </Grid>
-                    </ScrollSyncPane>
+        // we saved the start of this block of data, grab the information from there.
+        const orgName = props.data[offsetStartRow].organizationName;
+        const recvrName = props.data[offsetStartRow].receiverName;
+        const combinedName = `${orgName} ${recvrName}`.toLowerCase();
+        const successRate = Math.round(
+            (100 * successForRow.countSuccess) /
+                (successForRow.countSuccess + successForRow.countFailed)
+        );
+        const titleClassName =
+            SUCCESS_RATE_CLASSNAME_MAP[successForRow.currentState];
+
+        // if we have error text, then check if we filtered out all the slices
+        const allSlicesFiltered = filterErrorText.length
+            ? visibleSliceCount === 0
+            : false;
+
+        // cheat confession: using `data-` props allow us to stick properties
+        // on components without type checking requiring it be a formal prop.
+        perReceiverRowElements.push(
+            <Grid
+                row
+                key={`perreceiver-${combinedName}-${perReceiverRowElements.length}`}
+                className={"perreceiver-row"}
+                data-rowstatus={successForRow.currentState}
+                data-orgrecvname={combinedName}
+                data-allslicesfiltered={allSlicesFiltered}
+            >
+                <Grid className={`title-column ${titleClassName}`}>
+                    <div className={"title-text"}>
+                        {orgName}
+                        <br />
+                        {recvrName}
+                        <br />
+                        {successRate}%
+                    </div>
                 </Grid>
-            );
-        }
+                <ScrollSyncPane enabled>
+                    <Grid row className={"horizontal-scroll"}>
+                        <Grid row className={"week-column"}>
+                            {perDayElements}
+                        </Grid>
+                    </Grid>
+                </ScrollSyncPane>
+            </Grid>
+        );
         perDayElements = [];
+        visibleSliceCount = 0;
         // </editor-fold>
     } // while
+    return perReceiverRowElements;
+}
+
+/**
+ * Some filtering doesn't need to modify rows, just show/hide rows.
+ * This function loops over "pre-rendered" rows and decide which to
+ * include.
+ */
+function FilterRenderedRows(props: {
+    renderedRows: JSX.Element[];
+    filterRowStatus: SuccessRate;
+    filterRowReceiver: string;
+    filterErrorText: string;
+    onClick: (dataItem: AdmConnStatusDataType[]) => void;
+}) {
+    const filterRowStatus = props.filterRowStatus.toUpperCase(); // enum is uppercased
+    const filterRowReceiver = props.filterRowReceiver.toLowerCase();
```

Comment:
[nit] you have control over the values you're filtering against here, is there a reason not to just make the data look like you want it to look in the first place, rather than having to trim and change case here?

Reply:
There was, but I don't remember why. Clearly past me should have left a commented, I'll go try and fix and see what issue I hit.